Work with empty perf files

It should be possible to work with YAML files that are empty, it's sometimes easier to empty a file than delete it e.g. if you have it open in your text editor. For an empty file `yaml.load` returns `None` but we currently only accept dicts and thus this dies with:

```
        data = yaml.safe_load(content)
        if not isinstance(data, dict):
>           raise TypeError("YAML content of {} is not a dictionary".format(file_name))
E           TypeError: YAML content of foo.perf.yml is not a dictionary
```

--- a/HISTORY.rst
+++ b/HISTORY.rst
@@ -12,6 +12,7 @@ Pending release
   test records from calls outside of tests.
 * Made the automatic test detection work when running under a Pytest fixture.
 * Stopped throwing warnings on Python 3.
+* Fixed loading empty performance record files.
 
 2.0.1 (2017-03-02)
 ------------------

--- a/django_perf_rec/yaml.py
+++ b/django_perf_rec/yaml.py
@@ -14,6 +14,9 @@ class KVFile(object):
         self.file_name = file_name
         self.data = self.load(file_name)
 
+    def __len__(self):
+        return len(self.data)
+
     LOAD_CACHE = {}
 
     @classmethod
@@ -35,7 +38,10 @@ class KVFile(object):
                 raise
 
         data = yaml.safe_load(content)
-        if not isinstance(data, dict):
+
+        if data is None:
+            return {}
+        elif not isinstance(data, dict):
             raise TypeError("YAML content of {} is not a dictionary".format(file_name))
 
         return data

--- a/tests/test_yaml.py
+++ b/tests/test_yaml.py
@@ -29,6 +29,7 @@ class KVFileTests(SimpleTestCase):
 
     def test_load_non_existent_is_empty(self):
         kvf = KVFile(self.temp_dir + '/foo.yml')
+        assert len(kvf) == 0
         default = object()
         assert kvf.get('foo', default) is default
 
@@ -37,7 +38,23 @@ class KVFileTests(SimpleTestCase):
         with open(file_name, 'w') as fp:
             fp.write('foo: bar')
 
-        assert KVFile(file_name).get('foo', '') == 'bar'
+        kvf = KVFile(file_name)
+        assert len(kvf) == 1
+        assert kvf.get('foo', '') == 'bar'
+
+    def test_load_empty(self):
+        file_name = self.temp_dir + '/foo.yml'
+        with open(file_name, 'w') as fp:
+            fp.write('')
+
+        assert len(KVFile(file_name)) == 0
+
+    def test_load_whitespace_empty(self):
+        file_name = self.temp_dir + '/foo.yml'
+        with open(file_name, 'w') as fp:
+            fp.write(' \n')
+
+        assert len(KVFile(file_name)) == 0
 
     def test_load_non_dictionary(self):
         file_name = self.temp_dir + '/foo.yml'
@@ -52,6 +69,7 @@ class KVFileTests(SimpleTestCase):
         kvf = KVFile(self.temp_dir + '/foo.yml')
         kvf.set_and_save('foo', 'bar')
 
+        assert len(kvf) == 1
         assert kvf.get('foo', '') == 'bar'
 
     def test_load_second_same(self):
@@ -59,6 +77,7 @@ class KVFileTests(SimpleTestCase):
         kvf.set_and_save('foo', 'bar')
         kvf2 = KVFile(self.temp_dir + '/foo.yml')
 
+        assert len(kvf2) == 1
         assert kvf2.get('foo', '') == 'bar'
 
     def test_sets_dont_cause_append_duplication(self):

